src/character.h

```diff
@@ -543,9 +543,29 @@ class Character : public Creature, public visitable<Character>
 
         units::mass weight_carried() const;
         units::volume volume_carried() const;
+
+        /// Sometimes we need to calculate hypothetical volume or weight.  This
+        /// struct offers two possible tweaks: a collection of items and
+        /// coutnts to remove, or an entire replacement inventory.
+        struct item_tweaks {
+            item_tweaks() = default;
+            item_tweaks( const std::map<const item *, int> &w ) :
+                without_items( &w )
+            {}
+            item_tweaks( const inventory &r ) :
+                replace_inv( &r )
+            {}
+            const std::map<const item *, int> *without_items = nullptr;
+            const inventory *replace_inv = nullptr;
+        };
+
+        units::mass weight_carried_with_tweaks( item_tweaks ) const;
+        units::volume volume_carried_with_tweaks( item_tweaks ) const;
```

Comment:
Any reason these functions take by value and not by const reference?

Reply:
No particular reason; would work fine either way.